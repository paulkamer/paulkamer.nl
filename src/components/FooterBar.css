footer {
  background: #000;
  color: #fff;
  position: fixed;
  bottom: 0;
  width: 100vw;
  height: 3em;
  font-size: 0.8em;
  line-height: 3em;
}
