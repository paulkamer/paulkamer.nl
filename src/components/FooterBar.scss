@import "../styles/variables";

$height: 3em;
$icon-diameter: 20px;
$icon-padding: 0.5em;

.footerbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: $height;
  margin-top: auto;
  font-size: $font-size-smaller;
  line-height: $height;
  color: $color-black;
  background: $color-light-grey;

  &__section {
    display: flex;
    padding: 0 0.5em;
  }

  a {
    display: flex;
    color: $color-black;
  }

  &__icon {
    padding: 0 $icon-padding;

    img {
      display: inline-block;
      width: $icon-diameter;
      height: $icon-diameter;
    }
  }

  &__email {
    padding: 0 $icon-padding;
    font-size: $icon-diameter;
    font-weight: bold;
  }
}