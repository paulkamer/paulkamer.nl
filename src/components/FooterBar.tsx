import React from 'react';

import './FooterBar.scss';
import { lastUpdated } from '../data/last_updated';

const FooterBar: React.FC = () => {
  return (
    <footer className="footerbar">
      <div className="footerbar__section">
        <a
          href="https://github.com/paulkamer/"
          className="footerbar__icon"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            src="https://unpkg.com/simple-icons@latest/icons/github.svg"
            alt="Github"
          />
        </a>
        <a
          href="https://www.linkedin.com/in/paulkamer/"
          className="footerbar__icon"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            src="https://unpkg.com/simple-icons@latest/icons/linkedin.svg"
            alt="LinkedIn"
          />
        </a>
        <a
          href="mailto:paulkamer@gmail.com"
          className="footerbar__icon"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img src="/images/mail.svg" alt="email" />
        </a>
      </div>
      <div className="footerbar__section">
        last updated{' '}
        {lastUpdated.toLocaleDateString('en-us', {
          year: 'numeric',
          month: 'long'
        })}
      </div>
    </footer>
  );
};

export default FooterBar;
