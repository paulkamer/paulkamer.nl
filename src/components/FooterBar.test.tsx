import React from 'react';
import renderer from 'react-test-renderer';

import FooterBar from './FooterBar';

it('renders correctly', () => {
  const tree = renderer.create(<FooterBar />).toJSON();

  expect(tree).toMatchSnapshot();
});
