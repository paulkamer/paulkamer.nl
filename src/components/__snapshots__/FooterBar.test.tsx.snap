// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footerbar"
>
  <div
    className="footerbar__section"
  >
    <a
      className="footerbar__icon"
      href="https://github.com/paulkamer/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Github"
        src="https://unpkg.com/simple-icons@latest/icons/github.svg"
      />
    </a>
    <a
      className="footerbar__icon"
      href="https://www.linkedin.com/in/paulkamer/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="LinkedIn"
        src="https://unpkg.com/simple-icons@latest/icons/linkedin.svg"
      />
    </a>
    <a
      className="footerbar__icon"
      href="mailto:paulkamer@gmail.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="email"
        src="/images/mail.svg"
      />
    </a>
  </div>
  <div
    className="footerbar__section"
  >
    last updated Jan 2023
  </div>
</footer>
`;
